### PR TITLE
CLOUDP-67735: Handle ca certificate for digest transport

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-67735

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

We are not able to rely on the go client for setting the certificate, so I had to move some of the logic to the cli, I've tested this manually

### CODE REVIEW

File: internal/cli/opsmanager/ops_manager.go [hunk 1]
```diff
@@ -37,7 +37,11 @@ func Builder() *cobra.Command {
 		Short:   description.OpsManager,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 			config.SetService(config.OpsManagerService)
-			return validate.Credentials()
+			// do not validate to create an owner
+			if cmd.CommandPath() != "mongocli ops-manager owner create" {
```
Comment (gssbzn) [inline internal/cli/opsmanager/ops_manager.go:41]: drive by fix, owner is an unauth request

File: internal/store/store.go [hunk 2]
```diff
@@ -52,12 +54,32 @@ type Store struct {
...
+		DialContext: (&net.Dialer{
+			Timeout:   timeout,
+			KeepAlive: keepAlive,
+		}).DialContext,
+		MaxIdleConns:          maxIdleConns,
+		MaxIdleConnsPerHost:   maxIdleConnsPerHost,
+		Proxy:                 http.ProxyFromEnvironment,
+		IdleConnTimeout:       idleConnTimeout,
+		ExpectContinueTimeout: expectContinueTimeout,
+		TLSClientConfig:       tlsClientConfig,
```
Comment (gssbzn) [inline internal/store/store.go:76]: TL;DR: you can only set this on construction and the client was relying on reflection to "re set" this if needed, that reflection was failing when using the `digest.Transport` as the lib is agnostic about the authentication we use

File: internal/store/store.go [hunk 3]
```diff
@@ -69,22 +91,54 @@ func newTransport(c config.Config) *digest.Transport {
...
+}
+
+func defaultClient(c config.Config) (*http.Client, error) {
+	client := http.DefaultClient
+	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
+		dat, err := ioutil.ReadFile(caCertificate)
+		if err != nil {
+			return nil, err
+		}
+		fmt.Println(string(dat))
```
Comment (savannaendicott) [inline internal/store/store.go:126]: Was this used for dev/meant to be removed? Same with above on line 108?

File: internal/cli/owner/create.go [hunk 4]
```diff
@@ -98,7 +98,7 @@ func CreateBuilder() *cobra.Command {
 	cmd.Flags().StringVar(&opts.lastName, flag.LastName, "", usage.LastName)
 	cmd.Flags().StringSliceVar(&opts.whitelistIps, flag.WhitelistIP, []string{}, usage.WhitelistIps)
 
-	_ = cmd.MarkFlagRequired(flag.Username)
+	_ = cmd.MarkFlagRequired(flag.Email)
```
Comment (savannaendicott) [inline internal/cli/owner/create.go:101]: drive by fix?

Comment (gssbzn) [reply internal/cli/owner/create.go:101]: yep

File: internal/store/store.go (hunk 3 above)
Comment (gssbzn) [reply internal/store/store.go:126]: oops, thanks for the catch

**Comment by gssbzn** [comment]
@savannaendicott RFAL

**Pull request merged**